### ISSUE TITLE
loss and iou is nan in satellite_unit

### ISSUE THREAD

**Issue by Paranormaly** [karolzak/keras-unet#41]
I'm trying to train satellite_unit model but I got nan for loss, I tried your example on custome_unit it worked good on satellite images but when I'm trying satellite_unit i got this
here is the important part of the  code
`model = satellite_unet(
    input_shape,
)

callback_checkpoint = ModelCheckpoint(
    SAVED_MODEL,
    verbose=1,
    monitor='val_loss',
    save_best_only=True,
)

model.compile(
    optimizer=SGD(lr=0.01, momentum=0.99),
    loss='binary_crossentropy',
    # loss=jaccard_distance,
    metrics=[iou, iou_thresholded]
)

history = model.fit(
    train_gen,
    steps_per_epoch=200,
    epochs=5,
    validation_data=(x_val, y_val),
    callbacks=[callback_checkpoint]
)`

Here is the result for 5  epochs

`50/50 [==============================] - ETA: 0s - loss: nan - iou: nan - iou_thresholded: 3.8429
Epoch 00001: val_loss did not improve from inf
50/50 [==============================] - 251s 5s/step - loss: nan - iou: nan - iou_thresholded: 3.8429 - val_loss: nan - val_iou: nan - val_iou_thresholded: 1.6688e-07
Epoch 2/5
50/50 [==============================] - ETA: 0s - loss: nan - iou: nan - iou_thresholded: 1.3761e-07
Epoch 00002: val_loss did not improve from inf
50/50 [==============================] - 250s 5s/step - loss: nan - iou: nan - iou_thresholded: 1.3761e-07 - val_loss: nan - val_iou: nan - val_iou_thresholded: 1.6688e-07
Epoch 3/5
50/50 [==============================] - ETA: 0s - loss: nan - iou: nan - iou_thresholded: 1.4146e-07
Epoch 00003: val_loss did not improve from inf
50/50 [==============================] - 252s 5s/step - loss: nan - iou: nan - iou_thresholded: 1.4146e-07 - val_loss: nan - val_iou: nan - val_iou_thresholded: 1.6688e-07
Epoch 4/5
50/50 [==============================] - ETA: 0s - loss: nan - iou: nan - iou_thresholded: 1.3630e-07
Epoch 00004: val_loss did not improve from inf
50/50 [==============================] - 245s 5s/step - loss: nan - iou: nan - iou_thresholded: 1.3630e-07 - val_loss: nan - val_iou: nan - val_iou_thresholded: 1.6688e-07
Epoch 5/5
50/50 [==============================] - ETA: 0s - loss: nan - iou: nan - iou_thresholded: 1.3698e-07
Epoch 00005: val_loss did not improve from inf
50/50 [==============================] - 245s 5s/step - loss: nan - iou: nan - iou_thresholded: 1.3698e-07 - val_loss: nan - val_iou: nan - val_iou_thresholded: 1.6688e-07`

**Comment by karolzak** [comment]
Hi @Paranormaly 
Based on what you shared I would double check your input data.
I.e. I would check and validate what's the dtype and values range - both for X and Y

**Comment by Paranormaly** [comment]
Thanks @karolzak . the dtype in float32 here is the rest of the code . I used the default values trying to test the model

images = glob.glob("./DATASETS/AerialImageDataset/train/*.png")
masks = glob.glob("./DATASETS/AerialImageDataset/trainannot/*.png")

images_list = []
masks_list = []

for image, mask in zip(images, masks):
    images_list.append(np.array(Image.open(image).resize((384, 384))))
    masks_list.append(np.array(Image.open(mask).resize((384, 384))))

images_np = np.asarray(images_list)
masks_np = np.asarray(masks_list)`

x = np.asarray(images_np, dtype=np.float32) / 255
y = np.asarray(masks_np, dtype=np.float32)

y = y.reshape(y.shape[0], y.shape[1], y.shape[2], 1)
x_train, x_val, y_train, y_val = train_test_split(x, y, test_size=0.8, random_state=0)
train_gen = get_augmented(
    x_train, y_train,
    batch_size=2,
    data_gen_args=dict(
        rotation_range=5.,
        width_shift_range=0.05,
        height_shift_range=0.05,
        shear_range=40,
        zoom_range=0.2,
        horizontal_flip=True,
        vertical_flip=True,
        fill_mode='constant'
    ))
input_shape = x_train[0].shape
model = satellite_unet(
    input_shape,
)
`

**Comment by Paranormaly** [comment]
I found the cause it's   `y = np.asarray(masks_np, dtype=np.float32)`
it should be the same as X `y = np.asarray(masks_np, dtype=np.float32) / 255`
now the loss and iou is getting normal
